src/goto-symex/builtin_functions.cpp

```diff
@@ -355,6 +355,61 @@ void goto_symext::symex_printf(const expr2tc &, const expr2tc &rhs)
     cur_state->guard.as_expr(), cur_state->source, fmt.as_string(), args);
 }
 
+void goto_symext::symex_fscanf(
+  const expr2tc&,
+  const std::vector<expr2tc> &op)
+{
+  auto input_stream = op[0]; // stdin?
+  cur_state->rename(input_stream);
+  input_stream->dump();
+
+  /* TODO: Check for Format
+  const irep_idt fmt = get_string_argument(op[1]);
+  std::vector<size_t> alloc_size;
+  */
+
+  for(size_t i = 2; i < op.size(); i++)
+  {
+    // TODO: this should become a function
+    expr2tc operand = op[i];
+    internal_deref_items.clear();
+    dereference2tc deref(get_empty_type(), operand);
+    dereference(deref, dereferencet::INTERNAL);
+
+    for(const auto &item : internal_deref_items)
+    {
+      assert(is_symbol2t(item.object) && "This only works for variables");
+
+      // Get the length of the type. This will propagate an exception for dynamic/infinite
+      // sized arrays (as expected)
+      try
+      {
+        type_byte_size(item.object->type).to_int64();
+      }
+      catch(array_type2t::dyn_sized_array_excp *e)
+      {
+        log_error("no support VLAs");
+        abort();
```

Comment:
Shouldn't a (more) complete implementation support these as well? E.g.
```c
int arr[nondet_uint()];
fscanf(stdin, "%d", arr);
```
If so, please add a comment that it's just not implemented, yet. `"no support VLAs"` says nothing about why. I would also suggest to include "as fscanf parameter" in the message to help users.

Reply:
> Shouldn't a (more) complete implementation support these as well? 

The issue is that the stub is basically just converting it to nondet. In the case of arrays, should we make the entire array nondet or just one field?